src/server/hset_family.cc

```diff
@@ -695,6 +696,75 @@ OpResult<uint32_t> OpSet(const OpArgs& op_args, string_view key, CmdArgList valu
   return created;
 }
 
+struct OpSetExResult {
+  std::uint32_t created;
+  std::uint32_t updated;
+};
+
+OpResult<OpSetExResult> OpSetEx(const OpArgs& op_args, string_view key, CmdArgList values,
```

Comment:
What is the reason for spliting it from the OpSet function? is it because of the return value adding the updated filed? if so we can add it to OpSet and just ignore this return field when not needed in command return

Reply:
Notice that the resulting obj encoding is always StrMap.

So, we don't need to change `LpInsert` method for the ListPack, that is used [here](https://github.com/dragonflydb/dragonfly/blob/main/src/server/hset_family.cc#L670). 

Additionally, I removed all checks for the ListPack encoding from `OpSetEx`, like `op_sp.ttl == UINT32_MAX`.

Reply:
this are just code path that will not be executed in case of hsetex command. In this case I dont think we need to duplicate code

Reply:
In this case, we will have confusing contract for the `OpSet` method, won't we?

For the ListPack, number of the updated fields will always be 0, which will not correspond to reality. At the same time, for the StrMap, number of the updated fields can be bigger than 0.

Reply:
I assume Adi means that we can just extend OpSet with the NX option without copying all the code. If you need more info, you can tweak OpSet to return it and discard then irrelevant parts that don't belong to the reply